datafusion/core/tests/dataframe/mod.rs

```diff
@@ -1437,6 +1438,91 @@ async fn unnest_analyze_metrics() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn unnest_multiple_columns() -> Result<()> {
+    let df = table_with_mixed_lists().await?;
+    // Default behavior is to preserve nulls.
+    let results = df
+        .clone()
+        .unnest_columns(&["list", "large_list", "fixed_list"])?
+        .collect()
+        .await?;
+    let expected = [
+        "+------+------------+------------+--------+",
+        "| list | large_list | fixed_list | string |",
+        "+------+------------+------------+--------+",
+        "|      |            |            | d      |",
+        "|      |            | 3          | c      |",
+        "|      |            | 4          | c      |",
+        "|      | 2.2        | 1          | b      |",
+        "|      | 3.3        | 2          | b      |",
+        "|      | 4.4        |            | b      |",
+        "| 1    |            |            | a      |",
+        "| 2    | 1.1        |            | a      |",
+        "| 3    |            |            | a      |",
+        "+------+------------+------------+--------+",
+    ];
+    assert_batches_sorted_eq!(expected, &results);
```

Comment:
I think the result is deterministic, so we can check without sorting

Then, we can have a more straightforward result
```
[
    "+------+------------+------------+--------+",
    "| list | large_list | fixed_list | string |",
    "+------+------------+------------+--------+",
    "| 1    |            |            | a      |",
    "| 2    | 1.1        |            | a      |",
    "| 3    |            |            | a      |",
    "|      | 2.2        | 1          | b      |",
    "|      | 3.3        | 2          | b      |",
    "|      | 4.4        |            | b      |",
    "|      |            | 3          | c      |",
    "|      |            | 4          | c      |",
    "|      |            |            | d      |",
    "+------+------------+------------+--------+",
]
```

Reply:
> I think the result is deterministic, so we can check without sorting

Updated. Thank you @jayzhan211 . It looks much better now.